src/util/scene.ts

```diff
@@ -0,0 +1,161 @@
+import { Markup, MiddlewareFn, Scenes } from 'telegraf';
+import { BotContext } from '../BotContext';
+import { getProject, getResponse, handleError } from './botContext';
+import { Debugger } from 'debug';
+import { getProjectMembersFromString, isBackCommand } from './userInput';
+import { updateProjectInDb } from '../db/functions';
+import { InvalidTextError } from '../exceptions';
+
+export function makeSceneWithErrorHandling(
+    id: string,
+    debug: Debugger,
+    ...steps: MiddlewareFn<BotContext>[]
+) {
+    function addErrorHandlingToStep(
+        step: MiddlewareFn<BotContext>,
+    ): MiddlewareFn<BotContext> {
+        return async (ctx: BotContext, next: () => Promise<void>) => {
+            try {
+                return await step(ctx, next);
+            } catch (error) {
+                await handleError(ctx, error as Error, debug);
+                return ctx.wizard.selectStep(ctx.wizard.cursor);
+            }
+        };
+    }
+    return new Scenes.WizardScene<BotContext>(
+        id,
+        ...steps.map(addErrorHandlingToStep),
+    );
+}
```

Comment:
Felt that this abstraction helps a lot, especially with ```try catch``` statements that were previously repeated all the time. Perhaps a small suggestion would be to write this in the lambda function style, similar to the rest of the functions in ```scene.ts```